src/vs/workbench/services/workingCopy/common/workingCopyFileService.ts

```diff
@@ -272,37 +272,40 @@ export class WorkingCopyFileService extends Disposable implements IWorkingCopyFi
 	//#region File operations
 
 	create(operations: ICreateFileOperation[], undoInfo?: IFileOperationUndoRedoInfo, token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
-		return Promise.all(operations.map(o => this.doCreateFileOrFolder(o, true, undoInfo, token)));
+		return this.doCreateFileOrFolder(operations, true, undoInfo, token);
 	}
 
 	createFolder(operations: ICreateOperation[], undoInfo?: IFileOperationUndoRedoInfo, token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
-		return Promise.all(operations.map(o => this.doCreateFileOrFolder(o, false, undoInfo, token)));
+		return this.doCreateFileOrFolder(operations, false, undoInfo, token);
 	}
 
-	async doCreateFileOrFolder(operation: ICreateFileOperation | ICreateOperation, isFile: boolean, undoInfo?: IFileOperationUndoRedoInfo, token?: CancellationToken): Promise<IFileStatWithMetadata> {
+	async doCreateFileOrFolder(operations: (ICreateFileOperation | ICreateOperation)[], isFile: boolean, undoInfo?: IFileOperationUndoRedoInfo, token?: CancellationToken): Promise<IFileStatWithMetadata[]> {
 
 		// validate create operation before starting
 		if (isFile) {
-			const validateCreate = await this.fileService.canCreateFile(operation.resource, { overwrite: operation.overwrite });
-			if (validateCreate instanceof Error) {
-				throw validateCreate;
-			}
+			await Promise.all(operations.map(async operation => {
```

Comment:
I think you could also just await all those `canCreateFile` at once and check for the first `Error` in that array and throw that? Since in the code below you also bulk-create, I think it would be consistent.